spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/typeTests/AbstractTypeTest.scala

```diff
@@ -0,0 +1,427 @@
+
+    val readTableOptions = Map(
+      "keyspace" -> keyspaceName,
+      "table" -> typeNormalTable
+    )
+
+    val writeTableOptions = Map(
+      "keyspace" -> keyspaceName,
+      "table" -> s"${typeName}_dataframe"
+    )
+
+    val readDF = sqlContext
+      .read
+      .format(cassandraFormat)
+      .options(readTableOptions)
+      .load()
+
+    readDF
+      .write
+      .format(cassandraFormat)
+      .options(writeTableOptions)
+      .save()
+
+    val dataCopied = sc.cassandraTable[(TestType, TestType, TestType, TestType)](keyspaceName,s"${typeName}_dataframe").collect
+    checkNormalRowConsistency(typeData, dataCopied)
+  }
+
+  /**
+   * A method to insert expectedKeys which will be used during the typeTests test using the java
+   * driver
+   */
+  def insertData() = conn.withSessionDo { session =>
+
+    val normalStatement = session.prepare(
+      s"""INSERT INTO $keyspaceName.$typeNormalTable
+         |(pkey, ckey1, ckey2, data1)
+         |VALUES (?,?,?,?)""".stripMargin)
+    typeData.foreach { value: TestType =>
+      val driverValue = convertToDriverInsertable(value)
+      val bs = normalStatement.bind(driverValue, driverValue, driverValue, driverValue)
+      session.execute(bs)
+    }
+
+    val collectionStatement = session.prepare(
+      s"""INSERT INTO $keyspaceName.$typeCollectionTable
+         |(pkey, set1, list1, map1, map2)
+         |VALUES (?,?,?,?,?)""".stripMargin)
+    val scalaSet = typeSet.map(convertToDriverInsertable(_))
+    val javaSet: java.util.Set[DriverType] = scalaSet.asJava
+    val javaList: java.util.List[DriverType] = scalaSet.toList.asJava
+
+    val javaMap1: java.util.Map[String, DriverType] = typeMap.mapValues(value =>
+      convertToDriverInsertable(value)).asJava
+    val javaMap2: java.util.Map[DriverType, String] = typeReverseMap.map { case (key, value) =>
+      (convertToDriverInsertable(key), value)
+    }.asJava
+
+    typeData.foreach { value =>
+      val driverValue = convertToDriverInsertable(value)
+      val bs = collectionStatement.bind(driverValue, javaSet, javaList, javaMap1, javaMap2)
+      session.execute(bs)
+    }
+
+    val nullStatement = session.prepare(
+      s"""INSERT INTO $keyspaceName.${typeName}_null
+         |(pkey, data1, nulldata, nullset, nulllist, nullmap)
+         |VALUES (?,?,?,?,?,?)""".stripMargin)
+    typeData.foreach { value =>
+      val driverValue = convertToDriverInsertable(value)
+      val bs = nullStatement.bind(driverValue, driverValue, null, null, null, null)
+      session.execute(bs)
+    }
+  }
+
+  def generateKsTable() = conn.withSessionDo { session =>
+    val start = System.currentTimeMillis()
+    session.execute(s"""DROP KEYSPACE IF EXISTS $keyspaceName""")
+
+    session.execute(
+      s"""CREATE KEYSPACE IF NOT EXISTS $keyspaceName
+         |with replication = {'class':'SimpleStrategy','replication_factor':'1'}""".stripMargin)
+    session.execute(s"""USE $keyspaceName""")
+
+    session.execute(
```

Comment:
use `awaitAll(Future{...}, Future{...}, ...) to create independent schema elements